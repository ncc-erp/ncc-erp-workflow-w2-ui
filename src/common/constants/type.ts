export const InputTemplatesTypes = {
  OFFICE_LIST: "OfficeList",
  MY_PROJECT: "MyProject",
  TEXT: "Text",
  RICH_TEXT: "RichText",
  DATE_TIME: "DateTime",
};
