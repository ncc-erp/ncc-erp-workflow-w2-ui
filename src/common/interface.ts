export interface IPropertyDefinition {
  name: string;
  type: string;
  isRequired: boolean;
}

export interface IFormRequest {
  CurrentOffice: string;
  DestinationOffice: string;
  Content: string;
  StartDate: string;
  EndDate: string;
  Device: string;
  Project: string;
  Reason: string;
  Equipment: string;
  Dates: string;
}

export type FormField =
  | "CurrentOffice"
  | "DestinationOffice"
  | "Content"
  | "StartDate"
  | "EndDate"
  | "Device"
  | "Project"
  | "Reason"
  | "Equipment"
  | "Dates";
