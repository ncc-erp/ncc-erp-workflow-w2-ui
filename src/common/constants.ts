import { option } from 'common/types';

export const DEFAULT_FORMAT_DATE = 'dd-MM-yyyy HH:mm';
export const WFH_FORMAT_DATE = 'DD/MM/YYYY';

export const noOfRows: option[] = [
  {
    value: 10,
    label: 10,
  },
  {
    value: 25,
    label: 25,
  },
  {
    value: 50,
    label: 50,
  },
  {
    value: 100,
    label: 100,
  },
];

export const requestTemplateWorkflow = {
  CHANGE_OFFICE: 'Change Office Request',
  DEVICE_REQUEST: 'Device Request',
  OFFICE_EQUIPMENT: 'Office Equipment Request',
  WFH_REQUEST: 'WFH Request',
};

export const BoardColumnStatus = {
  Pending: 0,
  Approved: 1,
  Rejected: 2,
};

export const TaskStatus = {
  Pending: 0,
  Approved: 1,
  Rejected: 2,
};

export const FilterDate = {
  W1: '1 week',
  // W2: '2 weeks',
  // W3: '3 weeks',
  M1: '1 month',
  // M2: '2 months',
  M3: '3 months',
  // Y1: '1 years',
};

export const DislayValue = {
  BOARD: 0,
  LIST: 1,
};

export const UnitTime = {
  YEAR: 'years',
  MONTH: 'months',
  WEEK: 'weeks',
};

export const FilterAll = {
  USER: 'All users',
  DATE: 'All date',
  TYPE: 'All types',
  STATUS: 'All status',
};

export const UserRoles = {
  ADMIN: 'admin',
  DEFAULT_USER: 'DefaultUser',
};

export const UserAction = {
  CREATE: 'Create',
  EDIT: 'Edit',
  PERMISSIONS: 'Permissions',
  DELETE: 'Delete',
};

export const QueryKeys = {
  GET_ALL_ROLES: 'getAllRoles',
  GET_ROLE_BY_USER: 'getRoleByUserId',
  FILTER_USER: 'filterUser',
  FILTER_REQUEST: 'filterRequest',
  GET_REQUEST: 'getRequest',
  REQUEST_TEMPLATES: 'requestTemplates',
  GET_OFFICES: 'getOffices',
  GET_PROJECT_USER: 'getProjectsUser',
  GET_USER_INFO_WITH_BRANCH: 'getUserInfoWithBranch',
  GET_ALL_TASK: 'getAllTask',
  GET_ALL_TASK_FILTERED: 'getAllTaskFiltered',
  GET_TASK: 'getTask',
  FILTER_TASK: 'filterTask',
  GET_WFH_LIST: 'getWfhList',
  GET_USER_CURRENT_PROJECT: 'getUserCurrentProject',
  GET_STAKE_HOLDERS_FOR_FILTER: 'getStakeHoldersForFilter',
  GET_USER_LIST: 'getUserList',
};

export const ColorThemeMode = {
  LIGHT: 'light',
  DARK: 'dark',
};

export const DEFAULT_TASK_PER_PAGE = 10;
export const TaskStatusesData = [
  { status: 'Pending', color: 'blue' },
  { status: 'Approved', color: 'green' },
  { status: 'Rejected', color: 'red' },
  { status: 'Faulted', color: 'yellow' },
  { status: 'Cancelled', color: 'gray' },
];

export const OtherActionSignalStatus = {
  PENDING: 0,
  APPROVE: 1,
  REJECT: 2,
  FAILED: 3,
  CANCEL: 4,
};

export const ExternalAction = {
  APPROVED: 'approve',
  REJECTED: 'reject',
  OTHER: 'other',
};
