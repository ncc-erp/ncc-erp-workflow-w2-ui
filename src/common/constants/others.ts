export const formatsQuill = [
  "blockquote",
  "underline",
  "header",
  "italic",
  "strike",
  "bullet",
  "indent",
  "image",
  "bold",
  "list",
  "link",
];

export const modulesQuill = {
  toolbar: [
    [{ header: [1, 2, false] }],
    ["bold", "italic", "underline", "strike"],
    [{ list: "ordered" }, { list: "bullet" }],
    ["link", "image"],
    ["clean"],
  ],
};
