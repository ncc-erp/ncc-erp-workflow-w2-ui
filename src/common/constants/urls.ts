export const urls = {
  MY_REQUESTS: "/my-requests",
  REQUEST_TEMPLATES: "/request-templates",
  ACCOUNT: "/Account",
};
