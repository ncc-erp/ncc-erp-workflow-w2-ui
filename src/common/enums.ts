export enum LoginStatus {
  success = 1,
  failed = 2,
}

export enum RequestStatus {
  Approved = 'Approved',
  Rejected = 'Rejected',
  Pending = 'Pending',
  Canceled = 'Cancelled',
}

export enum ETaskStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
}

export enum RequestSortField {
  createdAt = 'createdAt',
  lastExecutedAt = 'lastExecutedAt',
}

export enum UserSortField {
  userName = 'userName',
  email = 'email',
  phoneNumber = 'phoneNumber',
}

export enum WfhSortField {
  email = 'Email',
  totalMissingPosts = 'totalMissingPosts',
}

export enum SortDirection {
  asc = 'asc',
  desc = 'desc',
}

export enum LocalStorageKeys {
  accessToken = 'accessToken',
  expireInSeconds = 'expireInSeconds',
  currentUser = 'currentUser',
  prevURL = 'prevURL',
}
