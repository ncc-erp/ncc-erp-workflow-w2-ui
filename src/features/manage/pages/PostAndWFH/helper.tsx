import { ColumnDef, createColumnHelper } from '@tanstack/react-table';
import { Box, Progress, Stack } from '@chakra-ui/react';
import { IPostAndWFH } from 'models/manage';

const columnHelper = createColumnHelper<IPostAndWFH>();

export const getPercentPostWFH = (
  numOfPosts: number,
  numOfRequestWFH: number
) => {
  return (numOfPosts / numOfRequestWFH) * 100;
};

export const postAndWTFColumns = [
  columnHelper.accessor('userId', {
    id: 'userId',
    header: () => <Box pl="16px">User ID</Box>,
    enableSorting: false,
    cell: (info) => <Box pl="16px">{info.getValue()}</Box>,
  }),
  columnHelper.accessor('email', {
    id: 'email',
    header: 'Email',
    enableSorting: false,
    cell: (info) => info.getValue(),
  }),
  columnHelper.accessor('numOfPosts', {
    id: 'numOfPosts',
    header: 'Number of posts',
    enableSorting: false,
    cell: (info) => info.getValue(),
  }),
  columnHelper.accessor('numOfRequestWFH', {
    id: 'numOfRequestWFH',
    header: 'Number of requests for WFH',
    enableSorting: false,
    cell: (info) => info.getValue(),
  }),
  columnHelper.display({
    id: 'status',
    enableSorting: false,
    header: () => <Box pr="16px">Status</Box>,
    cell: (info) => {
      const percent = getPercentPostWFH(
        info.row.original.numOfPosts,
        info.row.original.numOfRequestWFH
      );
      return (
        <Stack minW={200} pr="16px">
          <Progress
            w={'100%'}
            colorScheme={percent >= 100 ? 'green' : 'blue'}
            size="sm"
            value={percent}
          />
        </Stack>
      );
    },
  }),
] as ColumnDef<IPostAndWFH>[];
