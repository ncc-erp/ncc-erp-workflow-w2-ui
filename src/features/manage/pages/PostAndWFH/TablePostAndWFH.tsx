import {
  Box,
  Input,
  Center,
  HStack,
  Spacer,
  Spinner,
  InputGroup,
  InputRightElement,
} from '@chakra-ui/react';
import { useRecoilValue } from 'recoil';
import { TbSearch } from 'react-icons/tb';
import { noOfRows } from 'common/constants';
import useDebounce from 'hooks/useDebounce';
import { postAndWTFColumns } from './helper';
import usePostAndWFH from 'hooks/usePostAndWFH';
import { IFilterReportWFH } from 'models/manage';
import { appConfigState } from 'stores/appConfig';
import { Table } from 'common/components/Table/Table';
import { useCallback, useMemo, useState } from 'react';
import { Pagination } from 'common/components/Pagination';
import { PageSize } from 'common/components/Table/PageSize';
import { ShowingItemText } from 'common/components/Table/ShowingItemText';

const initialFilter: IFilterReportWFH = {
  search: '',
  pages: 0,
  per_page: +noOfRows[0].value,
};

export const TablePostAndWFH = () => {
  // State
  const [filter, setFilter] = useState<IFilterReportWFH>(initialFilter);
  const [email, setEmail] = useState<string>('');

  // Hooks
  const { sideBarWidth } = useRecoilValue(appConfigState);
  const { data, loading: isLoading, length } = usePostAndWFH(filter);

  const { pages, per_page } = filter;

  const currentPage = useMemo(
    () => (per_page + pages) / per_page,
    [pages, per_page]
  );

  const onPageChange = useCallback(
    (page: number) => {
      setFilter((filter) => ({
        ...filter,
        pages: filter.per_page * (page - 1),
      }));
    },
    [setFilter]
  );

  const onPageSizeChange = useCallback(
    (pageSize: number) => {
      setFilter((filter) => ({
        ...filter,
        per_page: pageSize,
        pages: 0,
      }));
    },
    [setFilter]
  );

  useDebounce(
    () =>
      setFilter((filter) => ({
        ...filter,
        search: email,
      })),
    300,
    [email]
  );

  return (
    <>
      <Box>
        <HStack
          w="full"
          pl="24px"
          pb="3px"
          alignItems="flex-end"
          flexWrap="wrap"
        >
          <InputGroup w={'20%'}>
            <Input
              type="text"
              placeholder="Enter email"
              fontSize="14px"
              mb={2}
              onChange={(e) => setEmail(e.target.value)}
            />
            <InputRightElement width="40px">
              <TbSearch />
            </InputRightElement>
          </InputGroup>
        </HStack>
        {isLoading ? (
          <Center h="200px">
            <Spinner mx="auto" speed="0.65s" thickness="3px" size="xl" />
          </Center>
        ) : (
          <Box
            p = "20px 30px 20px 30px"
            overflowX="auto"
            w={{ base: `calc(100vw - ${sideBarWidth}px)`, lg: 'auto' }}
          >
            <Table columns={postAndWTFColumns} data={data} />
          </Box>
        )}
        <HStack
          p = "0px 30px 20px 30px"
          justifyContent="space-between"
          borderBottom="1px"
          borderColor="gray.200"
          flexWrap="wrap"
        >
          <HStack alignItems="center" spacing="6px" flexWrap="wrap">
            <PageSize noOfRows={noOfRows} onChange={onPageSizeChange} />
            <Spacer w="12px" />
            <ShowingItemText
              skipCount={pages}
              maxResultCount={per_page}
              totalCount={length}
            />
          </HStack>
          <Pagination
            total={length}
            pageSize={per_page}
            current={currentPage}
            onChange={onPageChange}
            hideOnSinglePage
          />
        </HStack>
      </Box>
    </>
  );
};
