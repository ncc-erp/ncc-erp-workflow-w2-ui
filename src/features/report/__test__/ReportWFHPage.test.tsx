import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { RecoilRoot } from 'recoil';
import PostAndWFH from '..';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';
import { Permissions } from 'common/constants';

jest.mock('../../../api/apiHooks/index', () => ({
  useAxios: jest.fn(),
  useCreate: jest.fn(() => ({
    mutateAsync: jest.fn(),
    isLoading: false,
    isError: false,
  })),
}));

jest.mock('../../../api/axiosInstant', () => ({
  VITE_API_BASE_URL: '/api',
}));

jest.mock('hooks/useIsAdmin', () => ({
  useIsAdmin: jest.fn().mockReturnValue(true),
}));

jest.mock('hooks/useMediaQuery', () => ({
  useMediaQuery: jest.fn().mockReturnValue(true),
}));

jest.mock('api/apiHooks/reportHooks', () => ({
  useWfhList: jest.fn().mockReturnValue({
    isLoading: false,
    data: {
      totalCount: 1,
      items: [
        {
          branch: 'ĐN',
          creationTime: '2024-11-28T14:13:29.407344Z',
          email: 'manh.nguyenvan@ncc.asia',
          id: '3a16854b-80bf-aed5-4252-7be0b33d10d2',
          reason: 'Crudelis synagoga alias earum.',
          remoteDate: 20241215,
          status: 1,
        },
      ],
    },
  }),
}));

jest.mock('hooks/useUserPermissions', () => ({
  useUserPermissions: jest.fn(() => ({
    hasPermission: jest.fn(() => [Permissions.VIEW_WFH_REPORTS]),
  })),
}));

describe('WFH Page', () => {
  const totalCount: number = 1;
  test('should match snapshot when rendering', () => {
    const queryClient: QueryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <RecoilRoot>
          <Router>
            <PostAndWFH />
          </Router>
        </RecoilRoot>
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });

  describe('should behave as expected when total is 1', () => {
    const queryClient: QueryClient = new QueryClient();
    beforeEach(() => {
      render(
        <QueryClientProvider client={queryClient}>
          <RecoilRoot>
            <Router>
              <PostAndWFH />
            </Router>
          </RecoilRoot>
        </QueryClientProvider>
      );
    });

    it("should display the title 'Report WFH'", () => {
      expect(
        screen.getByRole('heading', { level: 1, name: /Report WFH/i })
      ).toBeInTheDocument();
    });

    it('should display the correct number of buttons on the screen', async () => {
      const buttonList = await screen.findAllByRole('button');
      expect(buttonList).toHaveLength(totalCount + 3);
    });

    it('should handle selecting rows per page', async () => {
      const options: string[] = ['10', '25', '50', '100'];
      for (const option of options) {
        const valueOption = screen.getByText(option);
        userEvent.click(valueOption);
        await screen.findByText(option);
      }
    });

    it('should display the search email', async () => {
      await screen.findByPlaceholderText('Enter email');
    });

    it('should display the input startDate', async () => {
      await screen.findByPlaceholderText('Start Date');
    });

    it('should display the input endDate', async () => {
      await screen.findByPlaceholderText('End Date');
    });
  });
});
