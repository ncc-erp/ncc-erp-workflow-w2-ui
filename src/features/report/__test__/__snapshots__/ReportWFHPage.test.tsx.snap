// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WFH Page should match snapshot when rendering 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-stack css-jrkth6"
    >
      <div
        class="chakra-stack__item css-1stbcqi"
      >
        <h1
          class="chakra-heading css-ftu039"
        >
          Report WFH
        </h1>
      </div>
      <div
        class="chakra-stack__item css-bdvyq0"
      >
        <div>
          <button
            class="chakra-button css-fkrxci"
            title="Release note"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="20px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.625 1.5H9a3.75 3.75 0 013.75 3.75v1.875c0 1.036.84 1.875 1.875 1.875H16.5a3.75 3.75 0 013.75 3.75v7.875c0 1.035-.84 1.875-1.875 1.875H5.625a1.875 1.875 0 01-1.875-1.875V3.375c0-1.036.84-1.875 1.875-1.875zm6.905 9.97a.75.75 0 00-1.06 0l-3 3a.75.75 0 101.06 1.06l1.72-1.72V18a.75.75 0 001.5 0v-4.19l1.72 1.72a.75.75 0 101.06-1.06l-3-3z"
                fill-rule="evenodd"
              />
              <path
                d="M14.25 5.25a5.23 5.23 0 00-1.279-3.434 9.768 9.768 0 016.963 6.963A5.23 5.23 0 0016.5 7.5h-1.875a.375.375 0 01-.375-.375V5.25z"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-fkrxci"
            title="User guide"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="20px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4 4a2 2 0 012-2h4.586A2 2 0 0112 2.586L15.414 6A2 2 0 0116 7.414V16a2 2 0 01-2 2H6a2 2 0 01-2-2V4zm2 6a1 1 0 011-1h6a1 1 0 110 2H7a1 1 0 01-1-1zm1 3a1 1 0 100 2h6a1 1 0 100-2H7z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <button
            class="chakra-button css-v0af4h"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
              />
            </svg>
            <p
              class="chakra-text css-14gt3bc"
            >
              Light
            </p>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-0"
      >
        <div
          class="chakra-stack css-12o5821"
        >
          <div
            class="chakra-input__group css-9zj08m"
            data-group="true"
          >
            <input
              class="chakra-input css-1vgf08c"
              placeholder="Enter email"
              type="text"
            />
            <div
              class="chakra-input__right-element css-yn8497"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <path
                  d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"
                />
                <path
                  d="M21 21l-6 -6"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-70qvj9"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                />
              </svg>
              <p
                class="chakra-text css-1gj3gif"
              >
                From
              </p>
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class=""
                    placeholder="Start Date"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <p
                class="chakra-text css-1etgquo"
              >
                To
              </p>
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="undefined undefined"
                    placeholder="End Date"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="chakra-wrap css-1md4fal"
        >
          <ul
            class="chakra-wrap__list css-1pgldbb"
          >
            <li
              class="chakra-wrap__listitem css-1yp4ln"
            >
              <div>
                <button
                  class="chakra-button css-4xx2wk"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                    />
                  </svg>
                </button>
              </div>
            </li>
            <li
              class="chakra-wrap__listitem css-1yp4ln"
            >
              <button
                aria-label="Done"
                class="chakra-button css-lp5hqr"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 0 1 655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 0 1 279 755.2a342.16 342.16 0 0 1-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 0 1 109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </div>
        <div
          class="css-1pf2gbx"
        >
          <table
            class="chakra-table css-zy166a"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-1y39i0r"
              >
                <th
                  class="css-j2saj0"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-13mxke4"
                  >
                    <div
                      class="css-0"
                    >
                      Email address
                    </div>
                  </div>
                </th>
                <th
                  class="css-j2saj0"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-13mxke4"
                  >
                    Number of requests for WFH
                  </div>
                </th>
                <th
                  class="css-j2saj0"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-13mxke4"
                  >
                    <svg
                      class="chakra-icon css-1pr7mfq"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M128 192l128 128 128-128z"
                      />
                    </svg>
                    Number of missing posts
                  </div>
                </th>
                <th
                  class="css-yddv3x"
                  colspan="1"
                  style="width: auto;"
                >
                  <div
                    class="css-u511cn"
                  >
                    <div
                      class="css-mgpx6q"
                    >
                      Actions
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="css-0"
            >
              <tr
                class="css-4g6ai3"
              >
                <td
                  class="css-jv3dxv"
                  style="word-break: break-word; white-space: pre-wrap;"
                >
                  <div
                    class="css-0"
                  >
                    van.nguyenhong@ncc.asia
                  </div>
                </td>
                <td
                  class="css-jv3dxv"
                  style="word-break: break-word; white-space: pre-wrap;"
                >
                  7
                </td>
                <td
                  class="css-jv3dxv"
                  style="word-break: break-word; white-space: pre-wrap;"
                >
                  <p
                    class="chakra-text css-12c7uv1"
                  >
                    5
                  </p>
                </td>
                <td
                  class="css-jv3dxv"
                  style="word-break: break-word; white-space: pre-wrap;"
                >
                  <div
                    class="css-gmuwbf"
                  >
                    <button
                      aria-controls="menu-list-:r5:"
                      aria-expanded="false"
                      aria-haspopup="menu"
                      aria-label=""
                      class="chakra-button chakra-menu__menu-button css-p7mkev"
                      id="menu-button-:r5:"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="chakra-icon css-bzk4kq"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.33409 4.54491C6.3494 3.63637 7.55145 2.9322 8.87555 2.49707C9.60856 3.4128 10.7358 3.99928 12 3.99928C13.2642 3.99928 14.3914 3.4128 15.1245 2.49707C16.4486 2.9322 17.6506 3.63637 18.6659 4.54491C18.2405 5.637 18.2966 6.90531 18.9282 7.99928C19.5602 9.09388 20.6314 9.77679 21.7906 9.95392C21.9279 10.6142 22 11.2983 22 11.9993C22 12.7002 21.9279 13.3844 21.7906 14.0446C20.6314 14.2218 19.5602 14.9047 18.9282 15.9993C18.2966 17.0932 18.2405 18.3616 18.6659 19.4536C17.6506 20.3622 16.4486 21.0664 15.1245 21.5015C14.3914 20.5858 13.2642 19.9993 12 19.9993C10.7358 19.9993 9.60856 20.5858 8.87555 21.5015C7.55145 21.0664 6.3494 20.3622 5.33409 19.4536C5.75952 18.3616 5.7034 17.0932 5.0718 15.9993C4.43983 14.9047 3.36862 14.2218 2.20935 14.0446C2.07212 13.3844 2 12.7002 2 11.9993C2 11.2983 2.07212 10.6142 2.20935 9.95392C3.36862 9.77679 4.43983 9.09388 5.0718 7.99928C5.7034 6.90531 5.75952 5.637 5.33409 4.54491ZM13.5 14.5974C14.9349 13.7689 15.4265 11.9342 14.5981 10.4993C13.7696 9.0644 11.9349 8.57277 10.5 9.4012C9.06512 10.2296 8.5735 12.0644 9.40192 13.4993C10.2304 14.9342 12.0651 15.4258 13.5 14.5974Z"
                        />
                      </svg>
                    </button>
                    <div
                      class="css-0"
                      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
                    >
                      <div
                        aria-orientation="vertical"
                        class="chakra-menu__menu-list css-1oaq8ss"
                        id="menu-list-:r5:"
                        role="menu"
                        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
                        tabindex="-1"
                      >
                        <button
                          class="chakra-menu__menuitem css-zn25ca"
                          data-index="0"
                          id="menu-list-:r5:-menuitem-:r6:"
                          role="menuitem"
                          tabindex="-1"
                          type="button"
                        >
                          <svg
                            class="chakra-icon css-zvw1k8"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 576 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                            />
                          </svg>
                          View
                        </button>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="chakra-stack css-9rshmm"
        >
          <div
            class="chakra-stack css-130amzp"
          >
            <div
              class="chakra-stack css-1hohgv6"
            >
              <p
                class="chakra-text css-1bevrez"
              >
                Rows per page
              </p>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <div
                  class="chakra-select__wrapper css-13voi5h"
                >
                  <select
                    class="chakra-select css-1y1ernp"
                    id="field-:r3:"
                  >
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      selected=""
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-10n8c1x"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon chakra-select__icon css-13otjrl"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      role="presentation"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: currentColor; width: 1em; height: 1em;"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 14L8 10H16L12 14Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-iz7l97"
            />
            <p
              class="chakra-text css-1bevrez"
            >
              showing 1 to 1 of 1 entries
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
